hoomd/md/force.py

```diff
@@ -46,39 +47,60 @@ class Force(_HOOMDBaseObject):
 
     @log(requires_run=True)
     def energy(self):
-        """float: Sum of the energy of the whole system \
+        """float: Total contribution to the potential energy of the system \
         :math:`[\\mathrm{energy}]`."""
         self._cpp_obj.compute(self._simulation.timestep)
         return self._cpp_obj.calcEnergySum()
 
     @log(category="particle", requires_run=True)
     def energies(self):
-        """(*N_particles*, ) `numpy.ndarray` of ``numpy.float64``: The \
-        energies for all particles :math:`[\\mathrm{energy}]`."""
+        """(*N_particles*, ) `numpy.ndarray` of ``float``: Energy \
+        contribution from each particle :math:`[\\mathrm{energy}]`."""
         self._cpp_obj.compute(self._simulation.timestep)
         return self._cpp_obj.getEnergies()
 
+    @log(requires_run=True)
+    def additional_energy(self):
+        """float: Additional energy term not included in `energies` \
+        :math:`[\\mathrm{energy}]`."""
+        self._cpp_obj.compute(self._simulation.timestep)
+        return self._cpp_obj.getExternalEnergy()
+
     @log(category="particle", requires_run=True)
     def forces(self):
-        """(*N_particles*, 3) `numpy.ndarray` of ``numpy.float64``: The \
-        forces for all particles :math:`[\\mathrm{force}]`."""
+        """(*N_particles*, 3) `numpy.ndarray` of ``float``: The \
+        force applied to each particle :math:`[\\mathrm{force}]`."""
         self._cpp_obj.compute(self._simulation.timestep)
         return self._cpp_obj.getForces()
 
     @log(category="particle", requires_run=True)
     def torques(self):
-        """(*N_particles*, 3) `numpy.ndarray` of ``numpy.float64``: The torque \
-        for all particles :math:`[\\mathrm{force} \\cdot \\mathrm{length}]`."""
+        """(*N_particles*, 3) `numpy.ndarray` of ``float``: The torque applied \
+        to each particle :math:`[\\mathrm{force} \\cdot \\mathrm{length}]`."""
         self._cpp_obj.compute(self._simulation.timestep)
         return self._cpp_obj.getTorques()
 
     @log(category="particle", requires_run=True)
     def virials(self):
-        """(*N_particles*, ) `numpy.ndarray` of ``numpy.float64``: The virial \
-        for all particles :math:`[\\mathrm{energy}]`."""
+        """(*N_particles*, 6) `numpy.ndarray` of ``float``: Virial tensor \
+        contribution from each particle :math:`[\\mathrm{energy}]`.
+
+        The 6 elements form the upper-triangular virial tensor in the order:
+        xx, xy, xz, yy, yz, zz.
+        """
         self._cpp_obj.compute(self._simulation.timestep)
         return self._cpp_obj.getVirials()
 
+    @log(category="sequence", requires_run=True)
+    def additional_virial(self):
+        """(1, 6) `numpy.ndarray` of ``float``: Additional virial tensor \
+        term not included in `virials` :math:`[\\mathrm{energy}]`."""
```

Comment:
Yes, these are terse. #966 is a reminder to fix this.

I didn't think `external` was the right name to expose to the user. This virial is not the result of external forces on the system, it arises from the interaction of the all the particles in the system with the periodic images with all the other particles in the system. Quantities like potential energy and virial are only defined in the sum over the system. The pair forces, for example, compute these per particle and adopt a certain convention on how much of each term is applied to the "per-particle" energy and virial. In PPPM, the energy and virial come from the FFTs, not the particles. 

We could choose to take this, divide it by N, and assign it to each particle. However, I took the path of the least amount of work and exported this extra term as an "additional" energy and virial. PPPM is the only code that uses it.

Reply:
I think the meaning of this will be clear after I complete #1078 which will explain that the total energy contributed to the system from a `Force` is `U = u_\mathrm{additional} \sum_{i=0}^{N} u_i` and similarly for the virial.